src/h5web/toolbar/controls/DomainSlider/utils.ts

```diff
@@ -1,8 +1,37 @@
-import type { CustomDomain, Domain } from '../../../vis-packs/core/models';
+import { extendDomain } from '../../../../packages/lib';
+import {
+  CustomDomain,
+  Domain,
+  ScaleType,
+} from '../../../vis-packs/core/models';
+import { createAxisScale } from '../../../vis-packs/core/utils';
 
-export function getSliderDomain(
+const EXTEND_FACTOR = 0.2;
+
+export function getVisDomain(
   dataDomain: Domain,
   customDomain: CustomDomain
 ): Domain {
   return [customDomain[0] ?? dataDomain[0], customDomain[1] ?? dataDomain[1]];
 }
+
+export function getSliderScale(domain: Domain, range: Domain) {
+  const sliderExtent = extendDomain(domain, EXTEND_FACTOR);
+
+  const scale = createAxisScale({
+    type: ScaleType.Linear,
+    domain: sliderExtent,
+    range,
+    round: true,
+  });
+
+  function scaleDomain(domain: Domain) {
+    return domain.map(scale) as Domain;
+  }
+
+  scaleDomain.invert = function scaleInvDomain(domain: Domain) {
+    return domain.map((val) => scale.invert(val)) as Domain;
+  };
```

Comment:
Now that the scale is abstracted in `ScaledSlider`, do you think it is worth keeping these functions ?

My point is that having `map` explicitly in `ScaledSlider` would be simpler to understand.

On the other hand, it is an implementation detail...

I won't push hard for any of the options :sweat_smile:

Reply:
Neh, you're right. Also, the scale no longer needs to be memoized now that I've removed the callbacks. The fact that I was extending the domain inside `useSliderScale` was a bit convoluted too... So I've removed the hook completely and inlined the domain extension, the scale creation and the `map`s in `SliderScale`.